### PR TITLE
Rewrite and review

### PR DESCRIPTION
**Pull request by fredrik-krokedil**
In accordance with [ClickUp task 1wgj5f7](https://app.clickup.com/t/1wgj5f7).

Refactored requests, rewrote most of meta-box and went through the rest of the code.

Please review for merge into develop branch.

### CODE REVIEW

File: classes/class-wc-klarna-order-management-order-lines.php [hunk 1]
```diff
@@ -22,42 +22,42 @@ class WC_Klarna_Order_Management_Order_Lines {
...
-	 * @var $order_tax_amount
+	 * @var integer
 	 */
 	public $order_tax_amount = 0;
 
 	/**
 	 * WooCommerce order ID.
 	 *
-	 * @var $order_id
+	 * @var string
```
Comment (MichaelBengtsson) [inline classes/class-wc-klarna-order-management-order-lines.php:46]: WooCommerce order IDs are ints generally. Is it a string here?

File: classes/request/class-kom-request.php [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+	 * The Klarna order id.
+	 *
+	 * @var string
+	 */
+	protected $klarna_order_id;
+
+	/**
+	 * The WC order id.
+	 *
+	 * @var string
```
Comment (MichaelBengtsson) [inline classes/request/class-kom-request.php:38]: As before, these are generally ints. Check if its a string here

**Review by MichaelBengtsson — COMMENTED** [review]
Simple stuff to check before QA

File: classes/class-wc-klarna-order-management-order-lines.php (hunk 1 above)
Comment (fredrik-krokedil) [reply classes/class-wc-klarna-order-management-order-lines.php:46]: The code this was originally called from, in the now removed includes/class-wc-klarna-order-management-request.php, it was declared as a string.
The same value now is in the KOM_Request class, and in trying to keep it as close to the original code as possible I declared it as string there as well.

Tracing it back through the request structure, into WooCommerce and out the other end, yes, it seems to be an int in the very bottom.

I'll update it _everywhere_ to reflect this.

**Comment by NiklasHogefjord** [comment]
Task linked: [CU-1wgj5f7 Rewrite of Klarna Order management](https://app.clickup.com/t/1wgj5f7)

**Pull request merged**